fgspectra/power.py

```diff
@@ -75,43 +75,43 @@ def __init__(self, filenames):
 
         for i, filename in np.ndenumerate(filenames):
             ell, spec = np.genfromtxt(filename, unpack=True)
-
+            ell = ell.astype(int)
             # Make sure that the new spectrum fits self._cl
             n_missing_ells = ell.max() + 1 - self._cl.shape[-1]
             if n_missing_ells > 0:
                 self._cl = np.pad(self._cl, ((0,0), (0, n_missing_ells)),
                                   mode='constant', constant_values=0)
 
-            self._cl[i+(ell,)] = spec
+            self._cl[(i,)+(ell,)] = spec
 
 
-    def __call__(self, ell):
+    def __call__(self, ell, ell_0=3000):
         """Compute the power spectrum with the given ell and parameters."""
-        return self._cl[..., ell]
+        return self._cl[..., ell] / self._cl[..., ell_0]
 
 
 class tSZ_150_bat(PowerSpectrumFromFile):
     """PowerSpectrum for Thermal Sunyaev-Zel'dovich (Dunkley et al. 2013)."""
 
     def __init__(self):
         """Intialize object with parameters."""
-        super().__init__(_get_power_file('tsz_150_bat'))
+        super().__init__([_get_power_file('tsz_150_bat')])
 
 
 class kSZ_bat(PowerSpectrumFromFile):
     """PowerSpectrum for Kinematic Sunyaev-Zel'dovich (Dunkley et al. 2013)."""
 
     def __init__(self):
         """Intialize object with parameters."""
-        super().__init__(_get_power_file('ksz_bat'))
+        super().__init__([_get_power_file('ksz_bat')])
 
 
 class sz_x_cib_template(PowerSpectrumFromFile):
```

Comment:
I think this spectrum should exist as a part of a correlated component, not on its own